### PR TITLE
Rename attributes() to getAttributes() to make it consistent

### PR DESCRIPTION
**Pull request by lukaszx0**
Make it consistent with everything else. Addresses https://github.com/grpc/grpc-java/issues/1779 and https://github.com/grpc/grpc-java/pull/2526#discussion_r96547749

/cc @ejona86 @dapengzhang0

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/ServerCallImpl.java [hunk 1]
```diff
@@ -192,7 +192,7 @@ ServerStreamListener newServerStreamListener(ServerCall.Listener<ReqT> listener)
   }
 
   @Override
-  public Attributes attributes() {
+  public Attributes getAttributes() {
     return stream.attributes();
```
Comment (lukaszx0) [inline core/src/main/java/io/grpc/internal/ServerCallImpl.java]: @ejona86 Heh, what about this one?

Comment (lukaszx0) [inline core/src/main/java/io/grpc/internal/ServerCallImpl.java]: There's also setter in this interface - https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/ServerStream.java#L83

At least it's in `internal` package, so I guess we should rename it to `getAttributes` too?

File: core/src/main/java/io/grpc/ServerCall.java [hunk 2]
```diff
@@ -237,7 +237,7 @@ public void setCompression(String compressor) {
    * @return Attributes container
    */
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1779")
-  public Attributes attributes() {
+  public Attributes getAttributes() {
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ServerCall.java]: Go ahead and have a `@Deprecated` and `@ExperimentalApi` `attributes()` method that calls the new method. That is very easy to do and makes it much easier to migrate.

File: core/src/main/java/io/grpc/internal/ServerCallImpl.java (hunk 1 above)
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/ServerCallImpl.java]: Meh. I'm not concerned either way.

File: core/src/main/java/io/grpc/ServerCall.java (hunk 2 above)
Comment (lukaszx0) [inline core/src/main/java/io/grpc/ServerCall.java]: Done.

File: core/src/main/java/io/grpc/internal/ServerCallImpl.java (hunk 1 above)
Comment (lukaszx0) [inline core/src/main/java/io/grpc/internal/ServerCallImpl.java]: I'll leave it that way then

Comment (lukaszx0) [inline core/src/main/java/io/grpc/internal/ServerCallImpl.java]: I've just discovered that `(Client|Server)Stream` are inconsistent in the same way, in which case I think it's worth adjusting: https://github.com/grpc/grpc-java/pull/2616

**Pull request merged**